src/codeu/chat/client/android/app/src/main/java/com/google/codeu/chatme/view/adapter/ChatListAdapter.java

```diff
@@ -0,0 +1,84 @@
+package com.google.codeu.chatme.view.adapter;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.google.codeu.chatme.view.ChatActivity;
+import com.google.codeu.chatme.R;
+import com.google.codeu.chatme.model.Conversation;
+import com.google.codeu.chatme.presenter.ChatActivityPresenter;
+
+import java.util.ArrayList;
+
+/**
+ * A {@link android.support.v7.widget.RecyclerView.Adapter} to bind the list of conversations
+ * data to the recyclerview in {@link ChatActivity}
+ */
+public class ChatListAdapter extends RecyclerView.Adapter<ChatListAdapter.ViewHolder>
+        implements ChatListAdapterView {
+
+    /**
+     * List of conversations
+     */
+    private ArrayList<Conversation> conversations = new ArrayList<>();
+
+    private ChatActivityPresenter presenter;
+
+    public ChatListAdapter() {
+        this.presenter = new ChatActivityPresenter(this);
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.chat_list_item, parent, false);
+        return new ViewHolder(v);
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        Conversation current = conversations.get(position);
+        holder.tvSender.setText(current.getOwner());
+    }
+
+    /**
+     * @return number of conversations
+     */
+    @Override
+    public int getItemCount() {
+        return conversations.size();
+    }
+
+    /**
+     * Loads conversations of the current user from Firebase database
+     */
+    public void loadConversations() {
+        this.presenter.loadConversations();
+    }
+
+    /**
+     * Resets the list of conversations and updates the recycler view
+     *
+     * @param conversations new list of conversations
+     */
+    public void setChatList(ArrayList<Conversation> conversations) {
+        this.conversations = conversations;
```

Comment:
I believe `notifyDataSetChanged()` might be needed to update the list of conversations (`RecyclerView`). The presenter (calling code) is delegating the adapter to update the view. What do you think?